ambry-router/src/main/java/com/github/ambry/router/NonBlockingRouter.java

```diff
@@ -646,23 +671,75 @@ public Future<Void> updateBlobTtl(RestRequest restRequest, String blobId, String
     Callback<Void> wrappedCallback =
         restRequest != null ? createIdConverterCallbackForTtlUpdateAndDelete(restRequest, blobId, futureResult,
             stringCallback) : callback;
+    if (restRequest == null) {
+      proceedWithTtlUpdate(blobId, serviceId, expiresAtMs, wrappedCallback, futureResult, quotaChargeCallback);
+    } else {
+      try {
+        idConverter.convert(restRequest, blobId, null, new Callback<String>() {
+          @Override
+          public void onCompletion(String convertedBlobId, Exception exception) {
+            if (exception != null) {
+              wrappedCallback.onCompletion(null, exception);
+            } else {
+              //make sure the blobId does not have the cluster prefix
+              proceedWithTtlUpdate(convertedBlobId, serviceId, expiresAtMs, wrappedCallback, futureResult,
+                  quotaChargeCallback);
+            }
+          }
+        });
+      } catch (Exception e) {
+        // Handle synchronous errors during header extraction
+        callback.onCompletion(null, e);
+        return futureResult;
+      }
+    }
+    return futureResult;
+  }
+
+  public String stripPrefixAndExtension(String blobId) throws RestServiceException {
+    return RestUtils.stripSlashAndExtensionFromId(
+        RequestPath.parse(blobId, Collections.emptyMap(), routerConfig.pathPrefixesToRemove, routerConfig.clusterName)
+            .getOperationOrBlobId(false));
```

Comment:
the blob id should be /xxxxx.bin if it's not named blob, in this case, we only have to do RestUtils.stripSlashAndExtentionFromId(blobId).

Reply:
You are right, fixed it.